iXBRLViewerPlugin/viewer/src/js/ixbrlviewer.js

```diff
@@ -161,6 +196,12 @@ iXBRLViewer.prototype._checkDocumentSetBrowserSupport = function () {
 iXBRLViewer.prototype.load = function () {
     var iv = this;
     var inspector = this.inspector;
+
+    // We need to parse JSON first so that we can determine feature enablement before loading begins.
+    const taxonomyData = iv._getTaxonomyData();
+    const parsedTaxonomyData = taxonomyData && JSON.parse(taxonomyData);
+    iv.setFeatures((parsedTaxonomyData && parsedTaxonomyData["features"]) || [], window.location.search);
```

Comment:
I understand the need to parse the JSON sooner, but does this need to be moved outside of the async setTimeout or can it move up earlier within it?

Reply:
I'll move it to the first thing within `setTimeout`. My concern was that if we add features going forward we may eventually need to know feature during initialization (before deferring via `setTimeout`). I suppose if that ever comes to fruition we could separate into two JSONs like `taxonomyData` and `config`, if delay from parsing large documents is the concern.